src/measure_theory/group.lean

```diff
@@ -0,0 +1,113 @@
+def is_left_invariant (μ : set G → ennreal) : Prop :=
+∀ (g : G) {A : set G} (h : is_measurable A), μ ((λ h, g * h) ⁻¹' A) = μ A
+
+/-- A measure `μ` on a topological group is right invariant if
+  for all measurable sets `s` and all `g`, we have `μ (sg) = μ s`,
+  where `sg` denotes the translate of `s` by right multiplication with `g`. -/
+def is_right_invariant (μ : set G → ennreal) : Prop :=
+∀ (g : G) {A : set G} (h : is_measurable A), μ ((λ h, h * g) ⁻¹' A) = μ A
+
+end
+
+namespace measure
+
+/-- The conjugate of a measure on a topological group.
+  Defined to be `A ↦ μ (A⁻¹)`, where `A⁻¹` is the pointwise inverse of `A`. -/
+protected def conj [measurable_space G] [group G] (μ : measure G) : measure G :=
+measure.map inv μ
+
+variables [measurable_space G] [group G] [topological_space G] [topological_group G] [borel_space G]
+
+lemma conj_apply (μ : measure G) {s : set G} (hs : is_measurable s) :
+  μ.conj s = μ s⁻¹ :=
+by { unfold measure.conj, rw [measure.map_apply measurable_inv hs, inv_preimage] }
+
+@[simp] lemma conj_conj (μ : measure G) : μ.conj.conj = μ :=
+begin
+  ext1 s hs, rw [μ.conj.conj_apply hs, μ.conj_apply, set.inv_inv], exact measurable_inv hs
+end
+
+variables {μ : measure G}
+
+lemma regular.conj [t2_space G] (h : μ.regular) : μ.conj.regular :=
+begin
+  split,
+  { intros K hK, rw [μ.conj_apply hK.is_measurable], apply h.le_top_of_is_compact,
+    exact (homeomorph.inv G).compact_preimage.mpr hK },
+  { intros A hA, rw [μ.conj_apply hA, ← h.outer_regular_eq],
+    refine le_of_eq _, apply infi_congr inv (equiv.inv G).injective.preimage_surjective,
+    intro U, apply infi_congr_Prop (homeomorph.inv G).is_open_preimage, intro hU,
+    apply infi_congr_Prop inv_subset_inv, intro h2U, rw [μ.conj_apply hU.is_measurable],
+    exact measurable_inv hA },
+  { intros U hU, rw [μ.conj_apply hU.is_measurable, ← h.inner_regular_eq],
+    refine ge_of_eq _, apply supr_congr inv (equiv.inv G).injective.preimage_surjective,
+    intro K, apply supr_congr_Prop (homeomorph.inv G).compact_preimage, intro hK,
+    apply supr_congr_Prop inv_subset_inv, intro h2U, rw [μ.conj_apply hK.is_measurable],
+    exact continuous_inv U hU },
+end
+
+end measure
+
+section conj
+variables [measurable_space G] [group G] [topological_space G] [topological_group G] [borel_space G]
+  {μ : measure G}
+
+@[simp] lemma regular_conj_iff [t2_space G] : μ.conj.regular ↔ μ.regular :=
+by { refine ⟨λ h, _, measure.regular.conj⟩, rw ←μ.conj_conj, exact measure.regular.conj h }
+
+lemma is_right_invariant_conj' (h : is_left_invariant μ) :
+  is_right_invariant μ.conj :=
+begin
+  intros g A hA, rw [μ.conj_apply (measurable_mul_right g hA), μ.conj_apply hA],
+  convert h g⁻¹ (measurable_inv hA) using 2,
+  simp only [←preimage_comp, ← inv_preimage],
+  apply preimage_congr, intro h, simp only [mul_inv_rev, comp_app, inv_inv]
+end
+
+lemma is_left_invariant_conj' (h : is_right_invariant μ) : is_left_invariant μ.conj :=
```

Comment:
Does something like `@is_right_invariant_conj' Gᵒᵖ ...` work?

Reply:
This requires a bunch of work (50+ lines) to even define the instances, which did not exist before.

I started on it in https://github.com/leanprover-community/mathlib/compare/haar-10, but I would like to make a separate PR for that.

Reply:
Sorry, I should've make it more clear that this is a question, not a request to make it work if it doesn't.